stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessageListViewModel.kt

```diff
@@ -236,6 +252,93 @@ public class MessageListViewModel(
         }
     }
 
+    /**
+     * Counts how many messages the user hasn't read already. This is based on what the last message they've seen is,
+     * and the current message state.
+     *
+     * @param newMessageState - The state that tells us if there are new messages in the list.
+     * @return - [Int] which describes how many messages come after the last message we've seen in the list.
+     * */
+    private fun getUnreadMessageCount(newMessageState: NewMessageState? = currentMessagesState.newMessageState): Int {
+        if (newMessageState == null || newMessageState == MyOwn) return 0
+
+        val messageItems = currentMessagesState.messageItems
+        val lastSeenMessagePosition =
+            getLastSeenMessagePosition(if (isInThread) lastSeenThreadMessage else lastSeenChannelMessage)
+        var unreadCount = 0
+
+        for (i in 0..lastSeenMessagePosition) {
+            val messageItem = messageItems[i]
+
+            if (!messageItem.isMine && messageItem.message.deletedAt == null) {
+                unreadCount++
+            }
+        }
+
+        return unreadCount
+    }
+
+    /**
+     * Gets the list position of the last seen message in the list.
+     *
+     * @param lastSeenMessage - The last message we saw in the list.
+     * @return - [Int] list position of the last message we've seen.
+     * */
+    private fun getLastSeenMessagePosition(lastSeenMessage: MessageItem?): Int {
+        if (lastSeenMessage == null) return 0
+
+        return currentMessagesState.messageItems.indexOfFirst {
+            it.message.id == lastSeenMessage.message.id
+        }
+    }
+
+    /**
+     * Attempts to update the last seen message in the channel or thread. We only update the last seen message the first
+     * time the data loads and whenever we see a message that's newer than the current last seen message.
+     *
+     * @param currentMessage - The message that is currently seen by the user.
+     * */
+    public fun updateLastSeenMessage(currentMessage: MessageItem) {
+        val lastSeenMessage = if (isInThread) lastSeenThreadMessage else lastSeenChannelMessage
+
+        if (lastSeenMessage == null) {
+            updateLastSeenMessageState(currentMessage)
+            return
+        }
+
+        val lastSeenMessageDate = lastSeenMessage.message.createdAt ?: Date()
```

Comment:
I have suggestion / enhancement

have a typealias for Date object  

As follows :  
_pseudo code from a scratch file_
```kotlin
import java.util.Date

typealias StreamDate = Date

data class MessageItem(
    val message : Message
)

data class Message(
    val createdAt : StreamDate?
)

lateinit var lastSeenMessage : MessageItem
val lastSeenMessageDate = lastSeenMessage.message.createdAt ?: Date()

// Suggestion
val lastSeenMessageDate = lastSeenMessage.message.createdAt ?: StreamDate()


// tomorrow we could
class SomeFancyDateLibClass {
    // Date logic
}
// go update 
typealias StreamDate = SomeFancyDateLibClass

```

Reply:
Pros : we simply update one typealias , entire code where we have Date imports gets reduced to one typealias

Reply:
To me that is very against the open-closed principle. A typealias is just a way to name a type differently, for more clarity, or cleaner syntax.

If we take this approach and we change the `typealias StreamDate = Date` to e.g. a date from the ThreeTen lib, all the underlying code/calling code would be broken as we'd have to completely change the call sites.

If we were to abstract it behind an interface, that would make more sense, but our parsers are using Date currently and we'd have to change the models too.

LMK if this makes sense or if you have other suggestions!